docs/source/using/host-folders.adoc

```diff
@@ -0,0 +1,225 @@
+Host folders allow two way file synchronization between host and VM.
+The following sections discuss the various types of host folders, driver provided host folders, as well as the Minishift `hostfolder` command.
+
+[[driver-host-folders]]
+== Driver provided host folders
+
+Some drivers mount a default host folder into the VM in order to share files between the VM and the host.
+These folders are currently not configurable and differ for each driver and OS.
+
+.Driver provided host folders
+[cols=",,,",options="header",]
+|========================================
+|Driver |OS |HostFolder |VM
+|Virtualbox |Linux |/home |/hosthome
+|Virtualbox |OSX |/Users |/Users
+|Virtualbox |Windows |C://Users |/c/Users
+|VMWare Fusion |OSX |/Users |/Users
+|Xhyve |OSX |/Users |/Users
+|========================================
+
+NOTE: Host folder sharing is not implemented in the KVM and Hyper-V drivers.
+If you use one of these drivers, you need to use <<mounting-custom-shared-folders, Minishift specific host folder options>>.
+
+NOTE: The VirtualBox default host folders are only available with the default Boot2Docker ISO.
+It requires VirtualBox link:https://www.virtualbox.org/manual/ch04.html[Guest Additions] to be available which is not the case for the CentOS ISO.
+
+[[minishift-host-folders]]
+== Minishift hostfolder command
+
+Minishift provides the link:../command-ref/minishift_hostfolder{outfilesuffix}[`hostfolder`] command to list, add, mount, unmount and remove host folders.
+In contrast to the <<driver-host-folders, driver provided host folders>>, the `hostfolder` command allows you to mount multiple shared folders onto custom specified mount points.
+
+At the moment only https://en.wikipedia.org/wiki/Server_Message_Block[CIFS] is supported as host folder type.
+Support for https://en.wikipedia.org/wiki/SSHFS[SSHFS] based host folders is in progress (see GitHub issue link:https://github.com/minishift/minishift/issues/317[#317]).
+If you want to use SSHFS already now, refer to <<sshfs-folder-mount>> for manual instructions.
+
+The following sections describe the prerequisites for using the `hostfolder` command, as well as the various sub-commands it provides.
+
+[[host-folder-prerequisites]]
+=== Prerequisites
+
+The prerequisite for using the `hostfolder` command is the ability to share directories via CIFS.
+On Windows CIFS is the default technology for sharing directories.
+For example, on Windows 10 the _C:\Users_ folder is shared by default and can be accessed by locally-authenticated users.
+
+It is also possible to use CIFS on OS X and Linux.
+On the former you can enable CIFS based shares via the  _System Preferences > Sharing_.
+Refer to link:https://support.apple.com/en-us/HT204445[How to connect with File Sharing on your Mac] for detailed setup instructions.
+
+On Linux follow your distribution specific instructions to install link:https://www.samba.org[Samba].
+
+[[displaying-host-folders]]
+=== Displaying host folders
```

Comment:
Suggest creating a Heading here, possibly: Minishift host folder subcommands and then listing all the sub commands at the 4th heading level.